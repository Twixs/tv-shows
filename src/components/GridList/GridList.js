import React from 'react';
import { Link } from 'react-router-dom';
import s from './GridList.module.scss';

export const GridList = ({ items }) => {
  if (!items || !items.length) {
    return <p>No items to display...</p>;
  }

  return (
    <div className={s.List}>
      {items.map((item) => (
        <Link to={`/shows/${item.id}`} key={item.id} className={s.ListItemLink}>
          <div className={s.ListItem}>
            <img src={item?.image?.medium} alt={item?.name} />
            <h3 className={s.ListItemTitle}>{item.name}</h3>
          </div>
        </Link>
      ))}
    </div>
  );
};
