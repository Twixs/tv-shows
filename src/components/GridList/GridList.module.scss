$darkPrimaryColor: #282c34;

.List {
  display: grid;
  grid-template-columns: repeat(auto-fill, 275px);
  grid-gap: 20px 0;
  justify-content: center;
  align-items: center;

  .ListItem {
    text-align: center;

    .ListItemTitle {
      margin-top: 10px;
    }
  }

  .ListItemLink {
    text-decoration: none;
    color: $darkPrimaryColor;
  }
}
