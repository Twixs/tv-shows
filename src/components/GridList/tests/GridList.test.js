import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { GridList } from '../GridList';
import renderer from 'react-test-renderer';

describe('GridList component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<GridList items={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should show appropriate message on empty list', () => {
    const { getByText } = render(<GridList items={[]} />);

    expect(getByText(/No items to display\.\.\./i)).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const history = createMemoryHistory();
    const items = [
      {
        id: 1,
        name: 'test',
      },
    ];
    const { getByText } = render(
      <Router history={history}>
        <GridList items={items} />
      </Router>
    );

    expect(getByText(/test/i)).toBeInTheDocument();
  });
});
