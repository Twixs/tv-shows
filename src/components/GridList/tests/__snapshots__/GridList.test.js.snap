// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridList component should match snapshot 1`] = `
<p>
  No items to display...
</p>
`;
